doc/GUIDE.md

```diff
@@ -0,0 +1,822 @@
+to jextract may result in weird errors due to missing macro definitions. A well-known
+example of this are Windows SDK headers. Almost always, the main `Windows.h` header file
+should be passed to jextract for things to work correctly. Please consult the
+documentation of the library that you're trying to use to find out which header file should
+be included/passed to jextract.
+
+## Command Line Option Reference
+
+A complete list of all the supported command line options is given below:
+
+| Option                                                       | Meaning                                                      |
+| :----------------------------------------------------------- | ------------------------------------------------------------ |
+| `-D --define-macro <macro>=<value>`                          | define `<macro>` to `<value>` (or 1 if `<value>` omitted)          |
+| `--header-class-name <name>`                                 | name of the generated header class. If this option is not specified, then header class name is derived from the header file name. For example, class "foo_h" for header "foo.h". |
+| `-t, --target-package <package>`                             | target package name for the generated classes. If this option is not specified, then unnamed package is used.  |
+| `-I, --include-dir <dir>`                                    | append directory to the include search paths. Include search paths are searched in order. For example, if `-I foo -I bar` is specified, header files will be searched in "foo" first, then (if nothing is found) in "bar".|
+| `-l, --library <name \| path>`                               | specify a shared library that should be loaded by the generated header class. If <libspec> starts with `:`, then what follows is interpreted as a library path. Otherwise, `<libspec>` denotes a library name. Examples: <br>`-l GL`<br>`-l :libGL.so.1`<br>`-l :/usr/lib/libGL.so.1`|
+| `--use-system-load-library`                                  | libraries specified using `-l` are loaded in the loader symbol lookup (using either `System::loadLibrary`, or `System::load`). Useful if the libraries must be loaded from one of the paths in `java.library.path`.| 
+| `--output <path>`                                            | specify where to place generated files                       |
+| `--dump-includes <String>`                                   | dump included symbols into specified file (see below)        |
+| `--include-[function,constant,struct,union,typedef,var]<String>` | Include a symbol of the given name and kind in the generated bindings. When one of these options is specified, any symbol that is not matched by any specified filters is omitted from the generated bindings. |
+| `--version`                                                  | print version information and exit      
+
+### Additional clang options
+
+Users can also specify additional clang compiler options, by creating a file named
```

Comment:
Maybe here we should start by saying that `jextract` uses clang to parse C files. And if additional options need to be passed to the clang instance used by jextract...

Reply:
I've added a leading sentence here in the latest version: `Jextract uses an embedded clang compiler (through libclang) to parse header files.`